lib/src/core/query_manager.dart

```diff
@@ -119,8 +119,11 @@ class QueryManager {
       ).first;
 
       // save the data from fetchResult to the cache
-      if (fetchResult.data != null &&
-          options.fetchPolicy != FetchPolicy.noCache) {
+      if (
+          // should never cache a mutation
```

Comment:
@truongsinh what's wrong with caching mutation results?

Reply:
Similar to HTTP POST request, it should never be cached, because a mutation is a request to the server to do something, not to get information (which is the functionality of a query)

Reply:
Hmm, I think pragmatically it's better to follow [apollo's design](https://www.apollographql.com/docs/react/advanced/caching#automatic-updates) regardless, but it also seems idiomatic according to the [graphql learn's mutation section](https://graphql.org/learn/queries/#mutations):
> Just like in queries, if the mutation field returns an object type, you can ask for nested fields. This can be useful for fetching the new state of an object after an update.

Reply:
Thanks for pointing out those 2 documents, frankly I haven't fully understood them, but for now it's safe to reverse this change (i.e. mutation can also be cached). Later on I will have some test to ensure that even with cache, the 2 uploads mutation are still call with 2 HTTP Multipart request